actors/miner/src/lib.rs

```diff
@@ -2002,56 +1937,26 @@ impl Actor {
             })?
             .ok_or_else(|| actor_error!(not_found, "no pre-commited sector {}", sector_number))?;
 
-        let max_proof_size = precommit.info.seal_proof.proof_size().map_err(|e| {
-            actor_error!(
-                illegal_state,
-                "failed to determine max proof size for sector {}: {}",
-                sector_number,
-                e
-            )
-        })?;
-        if params.proof.len() > max_proof_size {
+        let proof_lengths = vec![Some(params.proof.len())];
+        let precommits = vec![precommit.clone()];
+        let (batch_return, svis) = get_valid_verify_infos(rt, &precommits, &proof_lengths)?;
+        if batch_return.success_count != 1 {
             return Err(actor_error!(
                 illegal_argument,
-                "sector prove-commit proof of size {} exceeds max size of {}",
-                params.proof.len(),
-                max_proof_size
+                "failed to get an invalid verify info from precommit"
             ));
         }
-
-        let msd =
-            max_prove_commit_duration(rt.policy(), precommit.info.seal_proof).ok_or_else(|| {
-                actor_error!(
-                    illegal_state,
-                    "no max seal duration set for proof type: {:?}",
-                    precommit.info.seal_proof
-                )
-            })?;
-        let prove_commit_due = precommit.pre_commit_epoch + msd;
-        if rt.curr_epoch() > prove_commit_due {
+        let miner_actor_id: u64 = if let Payload::ID(i) = rt.message().receiver().payload() {
+            *i
+        } else {
             return Err(actor_error!(
-                illegal_argument,
-                "commitment proof for {} too late at {}, due {}",
-                sector_number,
-                rt.curr_epoch(),
-                prove_commit_due
+                illegal_state,
+                "runtime provided non ID receiver address {}",
```

Comment:
This won't happen. Just unwrap and let the panic cause USR_ASSERTION_FAILED, e..g in tests.

Fix the one in PCA too.

Reply:
I might be missing something but I don't think you can force a pattern match like this in rust.  We can circle back to this in a week if it turns out to be possible.